### PR TITLE
#78 Ability to create alarm on a dimension of a metric

### LINKED ISSUE #78: Ability to create alarm on a dimension of a metric
# This is a Feature Proposal

## Description
I collect metric data on types of transactions. The types are separated by name and version. These are defined as dimensions on the metric.

I want to alarm on if a particular name/version combination occurs. This seems to be possible through the aws cloudwatch cli. I see references to the dimensions flag in your code, but I can't locate any documentation on it being exposed through the plugin or how to use it if it is.

### PR DESCRIPTION
**Pull request by jandaluz**
## What did you implement:

Closes #78
Ability to create alarm on a dimension of a metric

## How did you implement it:

Adding the ability to parse a list of dimensions provided in the serverless yml for an alert. The FunctionName dimension will always be added by default, and be taken from the built cloud formation template.

## How can we verify it:

Run serverless deploy with dimensions list of key/values like mentioned in the README update. The generated alarm should be tagged with your custom dimensions in addition to the FunctionName

```yml
        dimensions:
          -  Name: foo
             Value: bar
```

## Todos:

- [ X ] Write tests
- [ X ] Write documentation
- [ X ] Fix linting errors
- [ X ] Provide verification config/commands/resources

### CODE REVIEW

**Comment by villekr** [comment]
If I understood this correctly then it would be possible to define alarms from e.g. AWS/ApiGateway metrics directly in your serverless.yaml. That would be exactly what I was already looking for. 
```
definitions:
  apigateway5xxError:
    description: 'myapi-5xxerrors'
    namespace: 'AWS/ApiGateway'
    metric: 5XXError
    threshold: 2
    statistic: Sum
    period: 300
    evaluationPeriods: 2
    comparisonOperator: GreaterThanOrEqualToThreshold
    treatMissingData: notBreaching
    Dimensions: 
      - Name: ApiName
      - Value: myapi
```

**Comment by maximivanov** [comment]
I came here with the same intent - wondering how do I add an alert for API Gateway's `5XXError` metric. Currently it's not possible while this PR solves it. The change doesn't seem to break anything. Please merge.

File: src/naming.js [hunk 1]
```diff
@@ -23,6 +23,24 @@ class Naming {
...
+      }
+    };
+    if(dimensionsList == null) {
+      return [funcNameDimension];
+    }
+    let filteredDimensions = dimensionsList.filter( (dim) => {
+      return dim.Name != 'FunctionName'
+    })
+    filteredDimensions.push(funcNameDimension);
+    console.log(filteredDimensions)
```
Comment (bulkan) [inline src/naming.js]: @jandaluz can you please remove the `console.log` ?

Comment (jandaluz) [reply src/naming.js]: i've pushed a new commit removing the log statement

**Pull request merged**